packages/order/src/migrations/Migration20240219102530.ts

```diff
@@ -0,0 +1,462 @@
+          order_change_id
+      );
+
+      CREATE INDEX IF NOT EXISTS "IDX_order_change_action_reference_id" ON "order_change_action" (
+          reference_id
+      );
+
+      CREATE TABLE IF NOT EXISTS "order_detail" (
+          "id" TEXT NOT NULL,
+          "order_id" TEXT NOT NULL,
+          "version" INTEGER NOT NULL,
+          "item_id" TEXT NOT NULL,
+          "quantity" NUMERIC NOT NULL,
+          "raw_quantity" JSONB NOT NULL,
+          "fulfilled_quantity" NUMERIC NOT NULL,
+          "raw_fulfilled_quantity" JSONB NOT NULL,
+          "shipped_quantity" NUMERIC NOT NULL,
+          "raw_shipped_quantity" JSONB NOT NULL,
+          "return_requested_quantity" NUMERIC NOT NULL,
+          "raw_return_requested_quantity" JSONB NOT NULL,
+          "return_received_quantity" NUMERIC NOT NULL,
+          "raw_return_received_quantity" JSONB NOT NULL,
+          "return_dismissed_quantity" NUMERIC NOT NULL,
+          "raw_return_dismissed_quantity" JSONB NOT NULL,
+          "written_off_quantity" NUMERIC NOT NULL,
+          "raw_written_off_quantity" JSONB NOT NULL,
+          "summary" JSONB NOT NULL,
+          "created_at" TIMESTAMPTZ NOT NULL DEFAULT Now(),
+          "updated_at" TIMESTAMPTZ NOT NULL DEFAULT Now(),
+          CONSTRAINT "order_detail_pkey" PRIMARY KEY ("id")
+      );
+
+      CREATE INDEX IF NOT EXISTS "IDX_order_detail_id" ON "order_detail" (id);
+
+      CREATE UNIQUE INDEX IF NOT EXISTS "IDX_order_detail_order_id_item_id_version" ON "order_detail" (
+          order_id,
+          item_id,
+          version
+      );
+
+      CREATE TABLE IF NOT EXISTS "order_line_item" (
+          "id" TEXT NOT NULL,
+          "totals_id" TEXT NULL,
+          "title" TEXT NOT NULL,
+          "subtitle" TEXT NULL,
+          "thumbnail" TEXT NULL,
+          "variant_id" TEXT NULL,
+          "product_id" TEXT NULL,
+          "product_title" TEXT NULL,
+          "product_description" TEXT NULL,
+          "product_subtitle" TEXT NULL,
+          "product_type" TEXT NULL,
+          "product_collection" TEXT NULL,
+          "product_handle" TEXT NULL,
+          "variant_sku" TEXT NULL,
+          "variant_barcode" TEXT NULL,
+          "variant_title" TEXT NULL,
+          "variant_option_values" JSONB NULL,
+          "requires_shipping" BOOLEAN NOT NULL DEFAULT true,
+          "is_discountable" BOOLEAN NOT NULL DEFAULT true,
+          "is_tax_inclusive" BOOLEAN NOT NULL DEFAULT false,
+          "compare_at_unit_price" NUMERIC NULL,
+          "raw_compare_at_unit_price" JSONB NULL,
+          "unit_price" NUMERIC NOT NULL,
+          "raw_unit_price" JSONB NOT NULL,
+          "created_at" TIMESTAMPTZ NOT NULL DEFAULT Now(),
+          "updated_at" TIMESTAMPTZ NOT NULL DEFAULT Now(),
+          CONSTRAINT "order_line_item_pkey" PRIMARY KEY ("id")
+      );
+
+      CREATE INDEX IF NOT EXISTS "IDX_order_line_item_variant_id" ON "order_line_item" (
```

Comment:
Do we ever fetch orders only by variant id? Or is it always product_id/variant_id combo? If it's the later, we can make this a composite index instead.

Reply:
These ids fall in the same category as things that can be expanded by other services via RemoteJoiner.